python/ray/tests/test_chaos.py

```diff
@@ -0,0 +1,138 @@
+import sys
+import random
+import string
+
+import ray
+
+import numpy as np
+import pytest
+import time
+
+from ray.data.impl.progress_bar import ProgressBar
+from ray._private.test_utils import get_all_log_message
+
+
+def assert_no_system_failure(p, total_lines, timeout):
+    # Get logs for 20 seconds.
+    logs = get_all_log_message(p, total_lines, timeout=timeout)
+    for log in logs:
+        assert "SIG" not in log, ("There's the segfault or SIGBART reported.")
+        assert "Check failed" not in log, (
+            "There's the check failure reported.")
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="Failing on Windows.")
+@pytest.mark.parametrize(
+    "ray_start_chaos_cluster", [{
+        "kill_interval": 3,
+        "timeout": 45,
+        "head_resources": {
+            "CPU": 0
+        },
+        "worker_node_types": {
+            "cpu_node": {
+                "resources": {
+                    "CPU": 8,
+                },
+                "node_config": {},
+                "min_workers": 0,
+                "max_workers": 4,
+            },
+        },
+    }],
+    indirect=True)
+def test_chaos_task_retry(ray_start_chaos_cluster, log_pubsub):
+    chaos_test_thread = ray_start_chaos_cluster
+    p = log_pubsub
+    chaos_test_thread.start()
+
+    # Chaos testing.
+    @ray.remote(max_retries=-1)
+    def task():
+        def generate_data(size_in_kb=10):
+            return np.zeros(1024 * size_in_kb, dtype=np.uint8)
+
+        a = ""
+        for _ in range(100000):
+            a = a + random.choice(string.ascii_letters)
+        return generate_data(size_in_kb=50)
+
+    @ray.remote(max_retries=-1)
+    def invoke_nested_task():
+        time.sleep(0.8)
+        return ray.get(task.remote())
```

Comment:
Do we need to turn on lineage reconstruction here? I think it is possible for the task to finish, then the node with the return object is killed right after.